src/js/cf-tables.js

```diff
@@ -0,0 +1,162 @@
+       * @returns function - A function to be used by the Array.sort method, where
+       * the parameters 'a' and 'b' is each an Array (of Arrays) to be sorted
+       */
+
+    function _arraySorter( sign, sortType ) {
+      return function( a, b ) {
+        // Set a and b to the first Array in each Array-of-Arrays
+        a = a[0];
+        b = b[0];
+
+        // For number sort, convert a & b to numbers.
+        if ( sortType === 'number' ) {
+          a = Number( a.replace(/[^\d.-]/g, '') );
+          b = Number( b.replace(/[^\d.-]/g, '') );
+        }
+
+        // Sort the values
+        if ( a < b ) {
+          return sign * -1;
+        }
+        if ( a > b ) {
+          return sign;
+        }
+        return 0;
+      };
+    }
+
+    /**
+     * Updates internal model of table (rows[])
+     * @param { index } - The index of the column used for sorting, which is
+     * used as the "key" for rows[] - it is set as the only value in the first
+     * array.
+     */
+    function _getRows( index ) {
+      // Clear the model
+      rows.length = 0;
+      // Find the value in each row of the column we're sorting by,
+      // add it to the rows Array
+      $table.find( 'tbody tr' ).each( function() {
+        // indices count from 0, but nth-child counts from 1
+        var child = index + 1,
+            key = $( this ).find( 'td:nth-child(' + child + ')' ).text().trim();
+        rows.push( [key, $( this )] );
+      });
+    }
+
+    /**
+     * Updates the table in the DOM
+     * @param { index } - The index of the column used for sorting
+     */
+    function _updateTable( index ) {
+      // Empty the tbody to prepare for sorted rows
+      $table.find( 'tbody' ).empty();
+
+      // Insert sorted rows
+      for( var i = 0; i < rows.length; i++ ) {
+        $table.find( 'tbody' ).append( rows[i][1] );
+      }
+    }
+
+    /**
+     * Handler for click events on the column header
+     * No parameters - uses SortableTable properties, updates the DOM.
+     */
+    function _clickHandler() {
+      $headercells.on( 'click', function() {
+        var sortType = $( this ).attr( 'data-sort_type' ),
+            sign = 1,
+            index = $table.find('tr:first-child').children( 'th, td' ).index( $( this ) );
+
+        _getRows( index );
+
+        // Handle .sortable__start-* classes
+        if ( $(this).hasClass( 'sortable__start-up' ) === true ) {
+          // Simply add the opposing sort class
+          $(this).addClass( 'sorted_down' );
+        }
+        else if ( $(this).hasClass( 'sortable__start-down' ) === true ) {
+          // Simply add the opposing sort class
+          $(this).addClass( 'sorted_up' );
+        }
+        $(this).removeClass( 'sortable__start-down sortable__start-up' );
+
+        // For reverse sorting, reverse the sign
+        if ( $(this).hasClass( 'sorted_up' ) === true ) {
+          sign = -1;
+          $( '.sortable' ).removeClass( 'sorted_up sorted_down' );
```

Comment:
I thought more about this, and this morning I realized that `sign` is all I really need to know to assign classes appropriately (note that `sign = 1` in the var declarations):

```
// Determine sign
if ( $cell.hasClass( 'sorted_up' ) || $cell.hasClass( 'sortable__start-down' ) ) {
  sign = -1;
}

// Remove existing sort classes in the table
$headercells.removeClass( 'sortable__start-down sortable__start-up' );
$headercells.removeClass( 'sorted_down sorted_up' );

// Add correct class
if ( sign === 1 ) {
  $cell.addClass( 'sorted_up' );
} 
else {
  $cell.addClass( 'sorted_down' );
}
```
(It also fixes the too-broad removal of classes from any `.sortable`). This feels pretty elegant. :smile: Thanks for the assist!